### PR TITLE
Re-factor integration tests to match JSON API naming structure

### PR DESCRIPTION
**Pull request by bravo-kernel**
I would appreciate feedback and/or approvals before moving forward with this.

## Proposal 1 (general)

I would like to start restructuring the integration tests so they adhere to the names/naming used on http://jsonapi.org/format/ to:

- make things more recognizable (everybody talking about the same jsonapi thing)
- improve maintenance (now almost impossible to find the relevant tests/fixtures)
- improve support (we could simply start pointing to working tests/fixtures)

## Proposal 2 (fixtures)

I also propose to move the fixtures into directories named after the tests (using exact same casing unless this gives problem). E.g.

### JsonApiRequestBodies

Integration tests should no longer use cake-array-to-json as request bodies since these could (and were giving) false positives. Moving fixtures to e.g.

- `Fixture/JsonApiRequestBodies/CreatingResources\*.json`
- `Fixture/JsonApiRequestBodies/UpdatingResources\*.json`

### JsonApiResponseBodies

To keep things recognizable these align with the JSON API naming:

- `Fixture/JsonApiResponseBodies/CreatingResources\*.json`
- `Fixture/JsonApiResponseBodies/FetchingCollections\*.json`
- `Fixture/JsonApiResponseBodies/FetchingResources\*.json`
- `Fixture/JsonApiResponseBodies/Sorting\*.json`
- `Fixture/JsonApiResponseBodies/SparseFieldsets\*.json`
- `Fixture/JsonApiResponseBodies/UpdatingResources\*.json`

### CODE REVIEW

**Comment by josegonzalez** [comment]
This seems reasonable. @AD7six I think you might have some thoughts on this, based on your crud/crud-json-api usage at work.

**Comment by bravo-kernel** [comment]
First error found (and solved) by not allowing cake-array-to-json as Request Body for integration tests anymore. https://github.com/FriendsOfCake/crud-json-api/pull/59/commits/56520568e6d8e397444d0eac1cf5cbe61da8d68d.

File: src/Listener/JsonApiListener.php [hunk 1]
```diff
@@ -67,7 +67,7 @@ class JsonApiListener extends ApiListener
      *
      * We attach at priority 10 so normal bound events can run before us
      *
-     * @return array
+     * @return mixed array or void
```
Comment (ADmad) [inline src/Listener/JsonApiListener.php]: nitpick: `void` shouldn't be mixed with any return type. Method should return `array` or `null`.

Comment (bravo-kernel) [reply src/Listener/JsonApiListener.php]: Thanks, never knew that. Updated as suggested.

**Comment by bravo-kernel** [comment]
OK, things look tidy and neat as far as I am concerned, with [this one exception](https://github.com/FriendsOfCake/crud-json-api/pull/59/files#diff-7a31363ea7896ebebf26bf77113ca48eR20).

Is it me or does the method name there look a bit odd? Suggestions to rename welcome.

Method found in https://github.com/bravo-kernel/crud-json-api/blob/refactor-tests/tests/TestCase/Integration/JsonApiBaseTestCase.php#L104.

**Comment by bravo-kernel** [comment]
This is good to merge. No tests were changed, only added.

**Pull request merged**